### DISCUSSION TITLE
API v2, Vibrator scene duration

### DISCUSSION THREAD

**Discussion by mmolotov** [zepp-health/zeppos-forum#239] [Developer Feedback]
Hi folks

according to the [DOC](https://docs.zepp.com/docs/reference/device-app-api/newAPI/sensor/Vibrator/) there is a list of constants to manage vibration duration.
i've tried all of them, but everytime result is the same - just one short vibration.

code examples:
```
import { Vibrator, VIBRATOR_SCENE_TIMER } from '@zos/sensor'

const vibrator = new Vibrator()
vibrator.start(VIBRATOR_SCENE_TIMER)
```

```
import { Vibrator, VIBRATOR_SCENE_TIMER } from '@zos/sensor'

const vibrator = new Vibrator()
vibrator.setMode(VIBRATOR_SCENE_TIMER)
vibrator.start()
```

i'm testing on T-Rex Ultra
any ideas what could be wrong?

thanks

**Comment by MyoungXUE** [comment]
Hi, this is confirmed to be a bug and we will fix it in the next OTA release!

**Reply by AlexandreFossati** [reply]
Did this release happened? I'm suffering the same problem today on Amazfit Bip 5 (zeppos v2.1)

**Reply by AlexandreFossati** [reply]
Well, I managed to make it work. I'm using the commands from the v2, but I took the approach needed from v1, which is to call `stop()` before `start()`.

Here's how the code looks like:
```js
import { Vibrator, VIBRATOR_SCENE_TIMER } from '@zos/sensor'

const vibrator = new Vibrator()
vibrator.stop()
vibrator.setMode(VIBRATOR_SCENE_TIMER)
vibrator.start()

setTimeout(() => {
  vibrator.stop()
}, 3000)
```

I want to suggest an update to the example from v2 documentation: https://docs.zepp.com/docs/v2/reference/device-app-api/newAPI/sensor/Vibrator/#example

**Comment by dejavu1987** [comment]
I am not getting any vibration on my T-Rex 2. 
I am trying to add vibration on `click_func`, I tried several different variations of `.start()` and `.setMode()` even the one from the latest comment `stop() -> setMode() -> start()`

**Reply by AlexandreFossati** [reply]
Are you sure you are importing the dependencies?
I struggled with it sometimes, maybe it's better to use typescript to not forget those silly details.
`import { Vibrator, VIBRATOR_SCENE_TIMER } from '@zos/sensor'`
